### PR TITLE
feat: scroll to top button

### LINKED ISSUE #3007: Scroll to top component
Since we added infinite scroll (#2859) it might be a good idea to add a new button that allows users to "scroll to the top" 

### Specs
- display at a certain page height
- fixed button bottom right
- reusable component
- mobile experience
- smooth animation 
- vanilla js

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3007 
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [x] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
   
image

### CODE REVIEW

**Review by petersopko — COMMENTED** [review]
wfm, but since @roiLeo opened the issue, I'm not quite sure about the specs/design here (I'd prefer the button to show up sooner + unfocused design could be a little bit more in style of KodaDot?)

File: components/rmrk/Gallery/ScrollTopButton.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  private scrollToTop = () => {
+    window.scroll({
+      top: 0,
+      behavior: 'smooth',
+    })
+  }
+}
+</script>
+<style lang="scss">
```
Comment (roiLeo) [inline components/rmrk/Gallery/ScrollTopButton.vue:39]: please make it scoped

File: components/rmrk/Gallery/ScrollTopButton.vue [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  <div v-if="showBtn" class="scroll-top-button" @click="scrollToTop">
+    <b-icon icon="arrow-alt-circle-up" />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'nuxt-property-decorator'
+import { Debounce } from 'vue-debounce-decorator'
+
+const SHOW_SCROLL_TOP_BUTTON_HEIGHT = 2000
```
Comment (roiLeo) [inline components/rmrk/Gallery/ScrollTopButton.vue:11]: I would use `350` and move it to `~/utils/constants`

**Review by roiLeo — COMMENTED** [review]
regarding design:
- icon size 24x24
- button size 56x56
- use `<a />` tag instead of `<div>` for semantic purposes

Can you move `ScrollTopButton` component into shared component folder since it will be used not only on rmrk chain

File: components/rmrk/Gallery/ScrollTopButton.vue (hunk 1 above)
Comment (Jarsen136) [reply components/rmrk/Gallery/ScrollTopButton.vue]: ok

File: components/rmrk/Gallery/ScrollTopButton.vue (hunk 2 above)
Comment (Jarsen136) [reply components/rmrk/Gallery/ScrollTopButton.vue]: ok

**Comment by Jarsen136** [comment]
Thanks for your review. I think it is better now.  
image

File: components/shared/ScrollTopButton.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <a
+    v-if="showBtn"
+    class="scroll-top-button is-flex is-align-items-center is-justify-content-center"
```
Comment (roiLeo) [inline components/shared/ScrollTopButton.vue:4]:
```suggestion
    class="scroll-top-button button is-rounded"
```

File: components/shared/ScrollTopButton.vue [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @click="scrollToTop">
+    <b-icon icon="chevron-up" class="is-24x24" />
```
Comment (roiLeo) [inline components/shared/ScrollTopButton.vue:6]:
```suggestion
    <b-icon icon="chevron-up" />
```

File: components/shared/ScrollTopButton.vue [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  private scrollToTop = () => {
+    window.scroll({
+      top: 0,
+      behavior: 'smooth',
+    })
+  }
+}
+</script>
+<style lang="scss" scoped>
+@import '@/styles/variables';
```
Comment (roiLeo) [inline components/shared/ScrollTopButton.vue:42]: this can be removed

```suggestion
```

File: components/shared/ScrollTopButton.vue [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  position: fixed;
+  right: 20px;
+  bottom: 100px;
+  width: 56px;
+  height: 56px;
+  z-index: 999999999;
+  &:hover {
+    color: $primary;
+    cursor: pointer;
+  }
```
Comment (roiLeo) [inline components/shared/ScrollTopButton.vue:54]: this can be removed
```suggestion
```

**Review by roiLeo — COMMENTED** [review]
last comments and we're good to launch

**Review by roiLeo — APPROVED** [review]
✅ lgtm, next step would be to wrap InfiniteLoading & ScrollTopButton components

**Comment by roiLeo** [comment]
pay 100

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 78.27 USD/KSM ~ 1.278 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xb9156fb0cfeb04b113001f6a05d96be49ef81bf503f4e92503fc445ce1d932a2](https://kusama.subscan.io/extrinsic/0xb9156fb0cfeb04b113001f6a05d96be49ef81bf503f4e92503fc445ce1d932a2)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**